### ISSUE TITLE
Usage with Playwright and Remix

### ISSUE THREAD

**Issue by Visible-Radio** [iFaxity/vite-plugin-istanbul#191]
Is there a way to use this with Playwright and Remix?

**Comment by iFaxity** [comment]
Hello @Visible-Radio, not sure i haven't tried Remix at all. Unsure what setup it uses, as this is a plugin for Vite it requires Vite at least. If it does use Vite then it should be able to work.

**Comment by Visible-Radio** [comment]
> Hello @Visible-Radio, not sure i haven't tried Remix at all. Unsure what setup it uses, as this is a plugin for Vite it requires Vite at least. If it does use Vite then it should be able to work.

Hey! Thanks for the reply. Remix does use Vite, but it also used esbuild. Although I guess vite _also_ uses esbuild under the hood. Generally this tooling stuff baffles me.

I tried following this example repo https://github.com/mxschmitt/playwright-test-coverage to get coverage information from my e2e tests, but was ultimately unsuccessful. I suspect it was due to some issue with the "instrumentation" step itself, but I can't say for sure.

**Comment by iFaxity** [comment]
Can you verify that the plugin is loaded? In the example you posted the env variable `USE_BABEL_PLUGIN_ISTANBUL` must be set for the plugin to be loaded.

**Comment by Visible-Radio** [comment]
It may not have been loaded.  Yesterday after posting I made some changes to my Remix/vite config (remix recently moved to vite from their own proprietary compiler - I hadn't fully migrated). Now I am getting _some kind_ of coverage results - but it doesn't seem to be correct.
I am invoking the tests like this
```shell
npx nyc npm run test:e2e 
```
Where the `test:e2e` script is just running `npx playwright test`.
Playwright starts the dev server as configured in my `playwright.config.ts`. Ultimately it invokes `remix vite:dev --manual`.

When everything starts up I see
```shell
[WebServer] vite:istanbul> Sourcemaps was automatically enabled for code coverage to be accurate.
To hide this message set build.sourcemap to true, 'inline' or 'hidden'.
```

Does this indicate the plugin is loaded?

**Comment by Visible-Radio** [comment]
FYI - there are a bunch of folks on the Remix discord that have asked about getting coverage out of Cypress / Playwright for their Remix apps, but there doesn't seem to be a clear-cut method to do it. It sounds like when they were on their own compiler it simply wasn't possible since there was no way to instrument it with Istanbul. 

I'm hoping their adoption of vite will make it possible - but I have no idea if there is some other magic in the remix build step that would make this difficult.

**Comment by iFaxity** [comment]
Yes the plugin seems to be loaded, otherwise the message would not have been printed.
I will try to run the remix example on how to get it to work with the plugin.

There are some confusing config that is needed for the plugin to run sometimes.
I am in the process of adding more logging as to if the plugin is correctly enabled or not #179.

**Comment by Visible-Radio** [comment]
Great! I'll stay tuned 🙂

**Comment by joeyjiron06** [comment]
@Visible-Radio, fwiw, i have a similar setup to you, but i'm using lit with vite. This plugin helped me get coverage. here's my setup that you can try to mimic with remix:


```js
// Playwright config
  webServer: {
    command: `vite --port ${port}`,
    port,
    ignoreHTTPSErrors: true,
    stdout: 'pipe',
    stderr: 'pipe',
    env: {
      VITE_COVERAGE: 'true', // used by vite-plugin-istanbul to enable coverage
    },
  },
```

```js
// vite.config.js
  plugins: [
    // ...
    istanbul({
      include: 'src/*',
      exclude: ['**/*.test.*', '**/*.spec.*'],
      extension: ['.js', '.ts'],
      requireEnv: true, // requires process.env.VITE_COVERAGE === 'true'
    }),
  ],
```

`.nycrc` in the root of my project
```json
{
  "extends": "@istanbuljs/nyc-config-typescript",
  "reporter": ["json"],
  "report-dir": "test-reports-e2e/coverage",
  "temp-dir": "test-reports-e2e/coverage",
  "include": "src/*",
  "exclude": ["**/*.test.*", "**/*.spec.*", "**/localization.ts"]
}
```

I installed [playwright-test-coverage](https://www.npmjs.com/package/playwright-test-coverage) for writing playwright tests and not having to add all the boilerplatte that the docs mentioned


```js
import { expect, test } from 'playwright-test-coverage';

// write tests here
```


```bash
# run playwright tests
nyc playwright test
```

**Comment by Visible-Radio** [comment]
Thanks @joeyjiron06. I'll have to dust off my experiments with this when I get a chance and see if your efforts can shed some light.

**Comment by cenfun** [comment]
@Visible-Radio Here is a example for generating Remix coverage with Playwright but it's using V8 not Istanbul.
https://github.com/cenfun/remix-with-playwright
